app/models/user.rb

```diff
@@ -9,7 +9,7 @@ class User < ApplicationRecord
 
   has_secure_password
 
-  validates :password, presence: true, length: {minimum: 6}
+  validates :password, presence: true, length: {minimum: 6}, allow_nil: true
```

Comment:
thêm allow_nil để làm gì? noi công dụng đê

Reply:
Ở đấy là để tránh báo trùng lỗi nhập mật khẩu trống khi đăng ký ạ
![err1](https://user-images.githubusercontent.com/32444642/31701107-279cfbb4-b3f8-11e7-9108-ab1c1504efae.png)

Reply:
có thật là để tránh lỗi nhập trống ko?

Reply:
À không, cái này là tránh trùng 2 lần lỗi Password can't be blank như e chụp ở trên ạ
1 lần là presence validation của password thì ta allow nil
1 lần là presence validation của has_secure_password thì chỉ bắt lỗi này thôi ạ

Reply:
Em thấy dùng allow_blank cx đk
Nên dùng allow_nil như thế này ko ạ